src/ui/ImageReference.hh

```diff
@@ -116,21 +119,30 @@ public:
             pRepository.ReleaseReference(*this);
         }
     }
-    
+
+private:
+    unsigned long GetData() noexcept { return GetDataImpl(*this); }
     /// <summary>
     /// Gets custom data associated to the reference - used to cache data by <see cref="ISurface::DrawImage" />.
     /// </summary>
-    unsigned long GetData() const noexcept { return m_nData; }
+    const unsigned long GetData() const noexcept { return GetDataImpl(*this); }
+    
+    template<typename T>
+    static auto GetDataImpl(T& ir) noexcept -> decltype(ir.GetData())
+    {
+        return ir.m_nData;
+    }
```

Comment:
This is overly complicated. Just have both `GetData` methods return `m_nData`.

Reply:
The non-const version can probably be eliminated.

Reply:
done